sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -308,7 +311,11 @@ private[hive] class SparkExecuteStatementOperation(
 object SparkExecuteStatementOperation {
   def getTableSchema(structType: StructType): TableSchema = {
     val schema = structType.map { field =>
-      val attrTypeString = if (field.dataType == NullType) "void" else field.dataType.catalogString
+      val attrTypeString = field.dataType match {
+        case NullType => "void"
+        case CalendarIntervalType => "string"
```

Comment:
Why is it string out of curiosity?

Reply:
cc @juliuszsompolski @HyukjinKwon
Maybe we need update `TCLIService.thrift`: https://issues.apache.org/jira/browse/HIVE-10037

Reply:
Hm. So we have thriftserver v1.2.1 imported, but don't have that patch released in v1.2.0?
Does it work with the v2.3.5 thriftserver?
It seems we indeed could import that HIVE patch instead of converting to string.

Reply:
No. Our interval type is different from Hive's Interval type.
Hive separate it to two types: `INTERVAL_YEAR_MONTH_TYPE`, `INTERVAL_DAY_TIME_TYPE`.

Reply:
Then I guess it has to stay a string for now, until Spark has iym and idt...